### DISCUSSION TITLE
[Enhancement] [input] ElInput maxlength，可否全局设置默认值

### DISCUSSION THREAD

**Discussion by skyshirt** [element-plus/element-plus#13281] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `input`

## 是否可以像渲染函数一样，使用表达示给组件设置默认值。
```vue
<script setup>
import { ElInput } from 'element-plus'
<script>
```
或是在main.ts中写入
```ts
import { ElInput } from 'element-plus'
ElInput.props.maxlength = '10'
// 或
ElInput.props.maxlength = {
  type: String,
  default: '10'
}
```
[函数式组件](https://cn.vuejs.org/guide/extras/render-function.html#functional-components)

### Additional comments

苦苦搜索，终未找到优雅合理的解决方案，希望大佬帮忙指导一下:smile:

**Comment by njh2wj** [comment]
相同问题

**Comment by btea** [comment]
基于你的需求，或许你可以对`input-number`封装一下，设置maxlength，然后在你的项目中使用封装的组件。

**Reply by skyshirt** [reply]
@btea 感谢回复！
其实封装个组件用不了多少时间，就是觉得对element-plus不太尊重。
印象里以前是可以配props.maxlength默认值的，为什么后来改成attrs了，而且这个attrs呃。。不知道怎么说，总感觉怪怪的。
不管怎么说，还是希望可以支持配置maxlength默认值:pray:

**Reply by btea** [reply]
我印象中似乎没有全局配置maxlength的操作。还是建议自己二次封装，不同的项目不同的开发者习惯要求不同，可统一配置属性太多了，开发者基于自己的项目进行封装配置是最灵活的选择。

maxlength配置似乎一直是这样的https://element-plus.org/en-US/component/input.html#limit-length 。你说的改成了attrs是指？

**Reply by skyshirt** [reply]
https://github.com/element-plus/element-plus/pull/3969 https://github.com/element-plus/element-plus/commit/3ade9cae65bd1b0ecdffe95a02846decfec406a6

**Comment by btea** [comment]
我之前还没注意到这个。 不过用法应该是一样的，原生input支持maxlength属性，所以直接把attrs绑定到input上是正确的。

**Reply by skyshirt** [reply]
是的，从表面看确实跟原来用法一致，可el-input也支持其它原生属性的props，比如placeholder disabled type readonly等，所以 其实 可能最开始我是带着负面情绪来建议的，那不好意思，打扰您的时间了。

**Reply by btea** [reply]
嗯，有疑问可以相互讨论。😀

**Reply by skyshirt** [reply]
领教！
